plugins/woocommerce/src/Internal/Admin/ProductForm/Field.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Handles product form field related methods.
+ */
+
+namespace Automattic\WooCommerce\Internal\Admin\ProductForm;
+
+/**
+ * Field class.
+ */
+class Field extends Component {
+	/**
+	 * Field type.
+	 *
+	 * @var string
+	 */
+	public $type;
+
+	/**
+	 * Array of required arguments.
+	 *
+	 * @var array
+	 */
+	const REQUIRED_ARGUMENTS = array(
+		'type',
+		'section',
+		'properties.name',
+		'properties.label',
+	);
+
+	/**
+	 * Constructor
+	 *
+	 * @param string $id Field id.
+	 * @param string $plugin_id Plugin id.
+	 * @param array  $additional_args Array containing the necessary arguments.
+	 *     $args = array(
+	 *       'type'            => (string) Field type. Required.
+	 *       'section'         => (string) Field location. Required.
+	 *       'order'           => (int) Field order.
+	 *       'properties'      => (array) Field properties.
```

Comment:
Just curious why we nest other arguments inside of properties.  Is this something we did in the previous API that we're following?

Reply:
We do something similar with the [`add_settings_fields`](https://developer.wordpress.org/reference/functions/add_settings_field/) with the `args` param.

I mostly wanted to differentiate what arguments would be specifically for the html field used as a property and which are more for general details like order/section/type etc....
But instead of relying on param order, I thought it was easier to pass them in as an argument array.